geonode/qgis_server/views.py

```diff
@@ -117,33 +119,14 @@ def download_qgs(request, layername):
 
     :return: QGS file.
     """
-
     layer = get_object_or_404(Layer, name=layername)
-    ogc_url = reverse('qgis_server:layer-request',
-                      kwargs={'layername': layername})
-    url = settings.SITEURL + ogc_url.replace("/", "", 1)
-
-    layers = [{
-        'type': 'raster',
-        'display': layername,
-        'driver': 'wms',
-        'crs': 'EPSG:4326',
-        'format': 'image/png',
-        'styles': '',
-        'layers': layer.title,
-        'url': url
-    }]
-
-    json_layers = json.dumps(layers)
-
-    url_server = settings.QGIS_SERVER_URL + \
-        '?SERVICE=PROJECTDEFINITIONS&LAYERS=' + json_layers
-    request = requests.get(url_server)
+    url = qgs_url(layer, internal=True)
+    request = requests.get(url)
     response = HttpResponse(
         request.content, content_type="application/xml",
         status=request.status_code)
     response['Content-Disposition'] = \
-        'attachment; filename=%s' % layer.title + '.qgs'
+        'attachment; filename=%s' % layer.name + '.qgs'
```

Comment:
Better to do it like this:

```
'attachment; filename=%s.qgs' % (layer.name, )
```

Also I actually prefer if you use `layer.title`. Because `layer.title` is usually more descriptive.
But if you use `layer.title` you should make sure that it works with filename like this:
`Flood hazard in Tanzania.qgs`

Reply:
Yes, layer tit le is better. Maybe just replace space by underscore?